src/main/java/seedu/duke/Parser.java

```diff
@@ -0,0 +1,135 @@
+package main.java.seedu.duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.regex.Pattern;
+
+import main.java.seedu.duke.commands.*;
+import main.java.seedu.duke.todo.Deadline;
+import main.java.seedu.duke.todo.Event;
+import main.java.seedu.duke.todo.Todo;
+import main.java.seedu.duke.todo.Task;
+
+/**
+ * Used for parsing strings for Duke project.
+ */
+public class Parser {
+    private static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM d yyyy");
+
+    /**
+     * Returns corresponding command specified in the input string.
+     * @param str The input string.
+     * @return Command.
+     * @throws DukeException If the input is incomplete.
+     */
+    public static Command parse(String str) throws DukeException {
+        String[] split1 = str.split(" ", 2);
+        String commandWord = split1[0].toLowerCase();
+        if (split1.length < 2 || split1[1].equals("")) {
+            // one word command
+            if (commandWord.equals("list")) {
+                return new ListCommand();
+            } else if (commandWord.equals("bye")) {
+                return new ExitCommand();
+            } else if (commandWord.equals("done")) {
+                throw new DukeException("Please tell me the task number that you have completed.");
+            } else if (commandWord.equals("date")) {
+                throw new DukeException("Please provide a date.");
+            } else if (commandWord.equals("delete")) {
+                throw new DukeException("Please tell me the task number that you want to delete.");
+            } else if (commandWord.equals("todo")) {
+                throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+            } else if (commandWord.equals("deadline")) {
+                throw new DukeException("OOPS!!! The description of a deadline cannot be empty.");
+            } else if (commandWord.equals("event")) {
+                throw new DukeException("OOPS!!! The description of a event cannot be empty.");
+            } else {
+                throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
```

Comment:
Perhaps this part should use switch case? And also for the todo/deadline/event, I think you can use String format to reduce the number of duplicate code